codegen/apipatcher.py

```diff
@@ -674,3 +675,41 @@ def _get_sub_structs(self, idl, structname):
             if structname2 in idl.structs:
                 structnames.update(self._get_sub_structs(idl, structname2))
         return structnames
+
+    @staticmethod
+    def get_structure_checks():
+        module = ast.parse(file_cache.read("backends/wgpu_native/_api.py"))
+        # We only care about top-level classes and their top-level methods.
+        top_level_methods = {
+            # (class_name, method_name) -> method_ast
+            (class_ast.name, method_ast.name): method_ast
+            for class_ast in module.body
+            if isinstance(class_ast, ast.ClassDef)
+            for method_ast in class_ast.body
+            if isinstance(method_ast, (ast.FunctionDef, ast.AsyncFunctionDef))
+        }
+
+        # (class_name, method_name) -> list of helper methods
+        method_helper_calls = defaultdict(list)
+        # (class_name, method_name) -> list of structures checked
+        structure_checks = defaultdict(list)
+
+        for key, method_ast in top_level_methods.items():
+            for node in ast.walk(method_ast):
+                if isinstance(node, ast.Call):
+                    name = ast.unparse(node.func)
+                    if name.startswith("self._"):
+                        method_helper_calls[key].append(name[5:])
+                    if name == "check_struct":
+                        if isinstance(node.args[0], ast.Constant):
```

Comment:
What would the `else` represent? From what I can see, `check_struct()` is always used with a string as first arg, so maybe we can assert this here?

Reply:
I try to be judicious in my use of "assert".  I prefer not to use it unless it really means a programmer error.  But I think you're right here.